websites/S/Stremio/presence.ts

```diff
@@ -20,133 +51,167 @@ presence.on("UpdateData", async () => {
 			presence.getSetting<boolean>("thumbnails"),
 			presence.getSetting<boolean>("buttons"),
 		]),
-		active = document.querySelector(
-			"[class='ng-binding ng-scope selected']"
-		)?.textContent;
+		sorted = document.querySelector('[title="Select catalog"]')?.textContent;
+
+	if (!strings) strings = await getStrings();
 
-	if (hostname.includes("app.strem.io")) {
-		search = document.querySelector("#global-search-field");
-		const video = document.querySelector<HTMLMediaElement>("#videoPlayer");
-		if (privacy && !video) presenceData.details = "Browsing...";
-		else if (privacy && video) presenceData.details = "Watching...";
-		else if (search?.value) {
-			presenceData.details = "Searching For:";
+	if (hostname === "web.stremio.com") {
+		search =
+			document.querySelector('[title="search videos"] > input') ||
+			document.querySelector('[placeholder="Search or paste link"]');
+		if (search?.value) {
+			presenceData.details = !privacy
+				? strings.searchFor
+				: strings.searchSomething;
+			presenceData.smallImageKey = Assets.Search;
 			presenceData.state = search.value;
-		} else if (hash.includes("/detail/")) {
-			title = document.querySelector(
-				"#detail > div:nth-child(3) > div > div.sidebar-info-container > div > div.logo > div"
-			).textContent;
-			presenceData.details = title;
-			presenceData.buttons = [
-				{
-					label: "Watch Video",
-					url: href,
-				},
-			];
-			if (thumbnails) {
-				presenceData.largeImageKey =
-					document
-						.querySelector(
-							"#detail > div.details-less-info > div.details-top > div:nth-child(1)"
-						)
-						?.firstElementChild.getAttribute("src") ?? "logo";
+			presence.setActivity(presenceData);
+			return;
+		}
+		switch (hash.split("/")[1]) {
+			case "addons": {
+				const type = lowerCaseIt(
+					document.querySelector('[title="Select type"]')?.textContent
+				);
+				presenceData.details = !privacy
+					? `Browsing ${lowerCaseIt(sorted).replace(" addons", "")} addons`
+					: "Browsing addons";
+				presenceData.state = type === "all" ? "" : `For ${type}`;
+				break;
+			}
+			case "settings": {
+				presenceData.details = !privacy
+					? `Viewing ${lowerCaseIt(
+							document.querySelector(
+								'[class*="side-menu-button-vbkJ1 selected-"]'
```

Comment:
you probs want to find a selector without that vbkJ1 as that is generated on each build and could be different in each... for the rest LGTM

Reply:
They are the same for all
![image](https://github.com/PreMiD/Presences/assets/42322979/7c9f33df-ca63-4045-a472-79a98696b67c)